File: src/main/java/com/jcs/suadeome/professionals/Professional.java
package com.jcs.suadeome.professionals;

import com.jcs.suadeome.types.Id;

public class Professional {

  private final Id id;
  private final String name;
  private final String phone;
  private final String service;

  public Professional(Id id, String name, String phone, String service) {
    this.id = id;
    this.name = name;
    this.phone = phone;
    this.service = service;
  }

  public Id getId() {
    return id;
  }

  public String getName() {
    return name;
  }

  public String getPhone() {
    return phone;
  }

  public String getService() {
    return service;
  }
}

File: src/main/kotlin/com/jcs/suadeome/professionals/Professional.kt
package com.jcs.suadeome.professionals

import com.google.gson.JsonElement
import com.google.gson.JsonObject
import com.google.gson.JsonSerializationContext
import com.google.gson.JsonSerializer
import com.jcs.suadeome.types.Id
import java.lang.reflect.Type

data class Professional(val id: Id, val name: String, val phone: PhoneNumber, val service: Service)

class ProfessionalJsonSerializer : JsonSerializer<Professional> {
    override fun serialize(src: Professional, typeOfSrc: Type, context: JsonSerializationContext): JsonElement {
        val obj = JsonObject();
        obj.addProperty("type", src.javaClass.simpleName)
        obj.addProperty("id", src.id.value)
        obj.addProperty("name", src.name)
        obj.addProperty("phone", src.phone.number)
        obj.addProperty("service", src.service.name)
        return obj
    }
}
